nixos/modules/services/misc/nix-daemon.nix

```
@@ -560,15 +560,17 @@ in
     environment.etc."nix/machines" =
       { enable = cfg.buildMachines != [];
         text =
-          concatMapStrings (machine:
-            "${if machine.sshUser != null then "${machine.sshUser}@" else ""}${machine.hostName} "
-            + (if machine.system != null then machine.system else concatStringsSep "," machine.systems)
-            + " ${if machine.sshKey != null then machine.sshKey else "-"} ${toString machine.maxJobs} "
+          concatMapStrings (machine: let
+            concatComma = lst: concatStringsSep "," lst;
+          in with builtins;
+            "${lib.optionalString (!isNull machine.sshUser) "${machine.sshUser}@"}${machine.hostName} "
```

```suggestion
            "${lib.optionalString (machine.sshUser != null) "${machine.sshUser}@"}${machine.hostName} "
```

Null isn't special when it comes to `==` and `!=`